commit 8728d7236a08191f206dfce7cd8f615a1c71a2ba
Author: Michael Still <mikal@stillhq.com>
Date:   Thu Aug 20 12:44:38 2026 +1000

    Install the kerbside-proxy wheel into kerbside images.
    
    Kerbside recently split out the proxying element which shovels SPICE
    packets around from the python business logic and re-implemented
    the high bandwidth proxy component in rust. The proxy is still pip
    installable because of a maturin wrapper, and the kerbside package
    depends on the kerbside-proxy package with a dev-inclusive floor
    which admits the rolling dev wheels published from the kerbside
    develop branch. Image builds which install kerbside from git
    therefore receive a matching proxy from PyPI with no help from
    kolla.
    
    However, it might be desirable for developers or deployers to be
    able to optionally override the kerbside-proxy wheel when building
    images. This patch therefore exposes such an override mechanism.
    
    Change-Id: Iaa556368a1d995a6764bd8b47833cc9472b1bb5d
    Signed-off-by: Michael Still <mikal@stillhq.com>

diff --git a/kolla/docker/kerbside/kerbside-base/Dockerfile.j2 b/kolla/docker/kerbside/kerbside-base/Dockerfile.j2
index bc181258f..fbf3fc519 100644
--- a/kolla/docker/kerbside/kerbside-base/Dockerfile.j2
+++ b/kolla/docker/kerbside/kerbside-base/Dockerfile.j2
@@ -35,6 +35,9 @@ RUN ln -s kerbside-source/* kerbside \
            {{ macros.install_pip(kerbside_plugins_pip_packages) }}; \
        fi \
     && {{ macros.install_pip(kerbside_pip_packages | customizable("pip_packages"), false) }} \
+    && if ls /kerbside/proxy-wheels/*.whl >/dev/null 2>&1; then \
+           {{ macros.install_pip(['/kerbside/proxy-wheels/*.whl'], false) }}; \
+       fi \
     && rm -f /etc/machine-id
 
 {{ macros.kolla_patch_sources() }}
